### ISSUE TITLE
In Python 3.7 `import collections` has a deprecation warning

### ISSUE THREAD

**Issue by annawinkler** [elastic/elasticsearch-dsl-py#1047]
Using this library with my application in Python 3.7, when I run tests, I get the following warning:
```
DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```

**Comment by annawinkler** [comment]
I created this PR to address this issue https://github.com/elastic/elasticsearch-dsl-py/pull/1046

**Comment by HonzaKral** [comment]
Thank you for the issue and the PR, merged as 193122c628cfa60769092de71e0d2e461dd2f142
